File: storm-core/src/test/java/st/orm/core/model/polymorphic/CharDiscAnimal.java
package st.orm.core.model.polymorphic;

import st.orm.DbTable;
import st.orm.Discriminator;
import st.orm.Discriminator.DiscriminatorType;
import st.orm.Entity;

/**
 * Single-Table Inheritance with CHAR discriminator type.
 */
@Discriminator(type = DiscriminatorType.CHAR)
@DbTable("char_disc_animal")
public sealed interface CharDiscAnimal extends Entity<Integer> permits CharDiscCat, CharDiscDog {
}

File: storm-kotlin/src/test/kotlin/st/orm/template/model/CharDiscAnimal.kt
package st.orm.template.model

import st.orm.DbTable
import st.orm.Discriminator
import st.orm.Discriminator.DiscriminatorType.CHAR
import st.orm.Entity
import st.orm.PK

@Discriminator(type = CHAR)
@DbTable("char_disc_animal")
sealed interface CharDiscAnimal : Entity<Int>

@Discriminator("C")
data class CharDiscCat(
    @PK val id: Int = 0,
    val name: String,
    val indoor: Boolean,
) : CharDiscAnimal

@Discriminator("D")
data class CharDiscDog(
    @PK val id: Int = 0,
    val name: String,
    val weight: Int,
) : CharDiscAnimal
